docs/dev/adding_instruments.rst

```diff
@@ -196,15 +196,16 @@ If, however, you are unable to use the :py:class:`~pymeasure.adapters.VISAAdapte
 
 .. code-block:: python
 
-    def __init__(self, resourceName, baud_rate=2400, **kwargs):
+    def __init__(self, adapter, baud_rate=2400, **kwargs):
         kwargs.setdefault('timeout', 0.5)
         kwargs.setdefault('xonxoff', True)
-        adapter = SerialAdapter(resourceName, 
+        kwargs.setdefault('name', "Extreme 5000")
+        adapter = SerialAdapter(adapter, 
```

Comment:
Here it's a bit more complicated, as `SerialAdapter` does not accept a `name` kwarg.
This should work correctly:
```suggestion
        name = kwargs.pop('name', "Extreme 5000")
        adapter = SerialAdapter(adapter, 
```
If the user has given a `name`, remove it from the dict and assign it to the variable. 
Below, use that `name` as the 2nd posarg instead of the string literal.

Reply:
I just realise: We should add a test to confirm correct behaviour for this usage pattern when the user passes in an `Adapter` instance instead of a string! I'm mad that I didn't think of that before. :-/

Reply:
> I just realise: We should add a test to confirm correct behaviour for this usage pattern when the user passes in an `Adapter` instance instead of a string! I'm mad that I didn't think of that before. :-/

I hope this does not invalidate this whole usage pattern :-(

Reply:
> > I just realise: We should add a test to confirm correct behaviour for this usage pattern when the user passes in an `Adapter` instance instead of a string! I'm mad that I didn't think of that before. :-/
> 
> I hope this does not invalidate this whole usage pattern :-(

Some quick tests indicate that we're fine for this :sweat_smile: -- Python uses the kwarg `name` element as the posarg called `name`, and in the Instrument initialiser it does not show up in that method's kwargs any more.
It's only problematic if an adapter is constructed inside the specific instrument's initialiser, as here and in one instrument below.

Reply:
> I just realise: We should add a test to confirm correct behaviour for this usage pattern when the user passes in an `Adapter` instance instead of a string! I'm mad that I didn't think of that before. :-/
> > 
> > I hope this does not invalidate this whole usage pattern :-(
> 
> Some quick tests indicate that we're fine for this 😅 -- Python uses the kwarg `name` element as the posarg called `name`, and in the Instrument initialiser it does not show up in that method's kwargs any more.

Yes, I also had tested, that a kwarg element supplants a posarg.

> It's only problematic if an adapter is constructed inside the specific instrument's initialiser, as here and in the ANC300 below.

What problem do you see? If the developer provides their own adapter, they have to either make the adapter take the `name` kwarg (just adding **kwargs is enough) or give a subset of the kwargs (without name) to their adapter.

> I just realise: We should add a test to confirm correct behaviour for this usage pattern when the user passes in an `Adapter` instance instead of a string! I'm mad that I didn't think of that before. :-/

Here again the developer should take care, if they modify the adapter behaviour.

Reply:
> What problem do you see? If the developer provides their own adapter, they have to either make the adapter take the name kwarg (just adding **kwargs is enough) or give a subset of the kwargs (without name) to their adapter.

Yes, only problematic for instrument driver developers that create custom adapters inside their instrument's init. The docs specify the correct behaviour, so I think we're good now.